src/sentry_plugins/bitbucket/endpoints/webhook.py

```diff
@@ -0,0 +1,225 @@
+                        #         type='D',
+                        #     )
+                        # for patched_file in path_set.modified_files:
+                        #     CommitFileChange.objects.create(
+                        #         organization_id=organization.id,
+                        #         commit=c,
+                        #         filename=patched_file.path,
+                        #         type='M',
+                        #     )
+                except IntegrityError:
+                    pass
+
+
+class BitbucketWebhookEndpoint(View):
+    _handlers = {
+        'repo:push': PushEventWebhook,
+    }
+
+    # https://developer.github.com/webhooks/
+    def get_handler(self, event_type):
+        return self._handlers.get(event_type)
+
+    def is_valid_signature(self, method, body, secret, signature):
+        if method == 'sha1':
+            mod = hashlib.sha1
+        else:
+            raise NotImplementedError('signature method %s is not supported' % (
+                method,
+            ))
+        expected = hmac.new(
+            key=secret.encode('utf-8'),
+            msg=body,
+            digestmod=mod,
+        ).hexdigest()
+        return constant_time_compare(expected, signature)
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, request, *args, **kwargs):
+        if request.method != 'POST':
+            return HttpResponse(status=405)
+
+        return super(BitbucketWebhookEndpoint, self).dispatch(request, *args, **kwargs)
+
+    def post(self, request, organization_id):
+        try:
+            organization = Organization.objects.get_from_cache(
+                id=organization_id,
+            )
+        except Organization.DoesNotExist:
+            logger.error('bitbucket.webhook.invalid-organization', extra={
+                'organization_id': organization_id,
+            })
+            return HttpResponse(status=400)
+
+        secret = OrganizationOption.objects.get_value(
+            organization=organization,
+            key='bitbucket:webhook_secret',
+        )
+        if secret is None:
+            logger.error('bitbucket.webhook.missing-secret', extra={
+                'organization_id': organization.id,
+            })
+            return HttpResponse(status=401)
+
+        body = six.binary_type(request.body)
+        if not body:
+            logger.error('bitbucket.webhook.missing-body', extra={
+                'organization_id': organization.id,
+            })
+            return HttpResponse(status=400)
+
+        try:
+            handler = self.get_handler(request.META['HTTP_X_EVENT_KEY'])
+        except KeyError:
+            logger.error('bitbucket.webhook.missing-event', extra={
+                'organization_id': organization.id,
+            })
+            return HttpResponse(status=400)
+
+        if not handler:
+            return HttpResponse(status=204)
+
+        # TODO(maxbittker) !!!validation is turned off here:
```

Comment:
hmmm from some cursory googling, it doesn't seem like bitbucket supports this: https://bitbucket.org/site/master/issues/12195/webhook-hmac-signature-security-issue

so we might need to talk to @mattrobenolt about alternatives